integration/auth_oidc_test.go

```diff
@@ -0,0 +1,302 @@
+package integration
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"strconv"
+	"testing"
+
+	"github.com/juanfont/headscale"
+	"github.com/juanfont/headscale/integration/dockertestutil"
+	"github.com/juanfont/headscale/integration/hsic"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+)
+
+const (
+	dockerContextPath      = "../."
+	hsicOIDCMockHashLength = 6
+	oidcServerPort         = 10000
+)
+
+var errStatusCodeNotOK = errors.New("status code not OK")
+
+type AuthOIDCScenario struct {
+	*Scenario
+
+	mockOIDC *dockertest.Resource
+}
+
+func TestOIDCAuthenticationPingAll(t *testing.T) {
+	IntegrationSkip(t)
+
+	baseScenario, err := NewScenario()
+	if err != nil {
+		t.Errorf("failed to create scenario: %s", err)
+	}
+
+	scenario := AuthOIDCScenario{
+		Scenario: baseScenario,
+	}
+
+	spec := map[string]int{
+		"namespace1": len(TailscaleVersions),
+	}
+
+	oidcConfig, err := scenario.runMockOIDC()
+	if err != nil {
+		t.Errorf("failed to run mock OIDC server: %s", err)
+	}
+
+	oidcMap := map[string]string{
+		"HEADSCALE_OIDC_ISSUER":                     oidcConfig.Issuer,
+		"HEADSCALE_OIDC_CLIENT_ID":                  oidcConfig.ClientID,
+		"HEADSCALE_OIDC_CLIENT_SECRET":              oidcConfig.ClientSecret,
+		"HEADSCALE_OIDC_STRIP_EMAIL_DOMAIN":         fmt.Sprintf("%t", oidcConfig.StripEmaildomain),
+		"HEADSCALE_ONLY_START_IF_OIDC_IS_AVAILABLE": "true",
```

Comment:
It should have `HEADSCALE_OIDC_` in front, and since it works it means its probably not needed?